app/models/manageiq/providers/azure/manager_mixin.rb

```diff
@@ -11,12 +11,23 @@ def connect(options = {})
 
   def verify_credentials(_auth_type = nil, options = {})
     require 'azure-armrest'
-    connect(options)
-  rescue Azure::Armrest::UnauthorizedException
-    raise MiqException::MiqHostError, _("Incorrect credentials - check your Azure Client ID and Client Key")
+    conf = connect(options)
+
+    unless subscription.blank?
+      vms = ::Azure::Armrest::VirtualMachineService.new(conf)
+      subscriptions = vms.subscriptions.map(&:subscription_id)
+      unless subscriptions.include?(subscription)
+        raise MiqException::MiqInvalidCredentialsError, _("Incorrect credentials - check your Azure Subscription ID")
+      end
+    end
+  rescue Azure::Armrest::UnauthorizedException, Azure::Armrest::BadRequestException
```

Comment:
Pretty sure. If you try `Azure::Armrest::ArmrestService.configure(cred)` with azure-armrest 0.2.x using invalid credentials, that is what you'll see.